ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java

```diff
@@ -30,31 +30,36 @@
 
   private static final String PROPERTY_PREFIX = "apiserver.";
 
-  public static final String VERTICLE_INSTANCES = propertyName("verticle-instances");
+  public static final String VERTICLE_INSTANCES = propertyName("verticle.instances");
   public static final int DEFAULT_VERTICLE_INSTANCES =
       2 * Runtime.getRuntime().availableProcessors();
   public static final String VERTICLE_INSTANCES_DOC =
       "The number of server verticle instances to start. Usually you want at least many instances"
           + " as there are cores you want to use, as each instance is single threaded.";
 
-  public static final String LISTEN_HOST = propertyName("listen-host");
+  public static final String LISTEN_HOST = propertyName("listen.host");
   public static final String DEFAULT_LISTEN_HOST = "0.0.0.0";
   public static final String LISTEN_HOST_DOC =
       "The hostname to listen on";
 
-  public static final String LISTEN_PORT = propertyName("listen-port");
+  public static final String LISTEN_PORT = propertyName("listen.port");
   public static final int DEFAULT_LISTEN_PORT = 8089;
   public static final String LISTEN_PORT_DOC =
       "The port to listen on";
 
-  public static final String KEY_PATH = propertyName("key-path");
+  public static final String KEY_PATH = propertyName("key.path");
   public static final String KEY_PATH_DOC =
       "Path to key file";
 
-  public static final String CERT_PATH = propertyName("cert-path");
+  public static final String CERT_PATH = propertyName("cert.path");
   public static final String CERT_PATH_DOC =
       "Path to cert file";
 
+  public static final String WORKER_POOL_SIZE = propertyName("worker.pool.size");
+  public static final String WORKER_POOL_DOC =
+      "Max number of worker threads for executing blocking code";
+  public static final int DEFAULT_WORKER_POOL_SIZE = 100;
```

Comment:
Why does it make sense to have so many workers relative to the number of verticles? Is the expectation that multiple blocking tasks created by the same will be running simultaneously?

Reply:
We choose the number of event loops to be about the same as the number of cores because event loops should always be live, therefore you shouldn't need any more to utilise all the cores.
Workers are often used to execute blocking tasks - often blocking on IO, e.g. waiting on the network or file system etc. In this case the threads can spend a lot of their time inactive so in order to utilise all cores we choose a much larger number of threads than number of cores.
In our case if we find our workers aren't blocking on IO much, just doing long lived CPU tasks then we can probably get away with fewer, although we probably still want a reasonable amount so we don't end up with very long lived tasks causing those behind them to queue too long.